### PR TITLE
Do not use ToString if flag is not set

### PR DESCRIPTION
**Pull request by BenjaminTrapani**
Bug fix after testing this change in gillnet

### CODE REVIEW

**Comment by BenjaminTrapani** [comment]
@jamesgarfield @quintenpalmer Could you please review these changes? Hopefully this is the last set. A useful addition to our vendoring process would be the ability to vendor a branch of an existing repo in order to test changes before opening a PR.

**Comment by quintenpalmer** [comment]
Mm, that would be nice, agreed @BenjaminTrapani. I think I remember cheating and building jsonenums from my clone of this repo and running the binary myself on the packages I wanted, to test that it was behaving as I expected before posting a PR for a change and merging it. I may be misremembering because I also remember posting multiple PRs against jsonenums myself.

**Review by quintenpalmer — APPROVED** [review]
Changes LGTM, maybe we could add a good set of tests to this in a separate ticket/epic?

**Comment by BenjaminTrapani** [comment]
@quintenpalmer Definitely, probably just one ticket to test jsonenums effectively. Should I log a ticket to do so?

**Comment by quintenpalmer** [comment]
@BenjaminTrapani that would be great; we could advocate for that in arch, I doubt we'll get much opposition.

File: template.go [hunk 1]
```diff
@@ -57,7 +57,15 @@ func (r {{$typename}}) getString() (string, error) {
     if s, ok := interface{}(r).(fmt.Stringer); ok {
         return s.String(), nil
     }
+    {{if $.Stringer}}
     return r.ToString()
+    {{else}}
+    s, ok := _{{$typename}}ValueToName[r]
```
Comment (jamesgarfield) [inline template.go]: Why are we duplicating this code between here and `ToString` instead of extracting it to a common private function they can both use?

**Comment by jamesgarfield** [comment]
govendor has a way to vendor from a specified commit. From their help text the following is the pack specifier grammar
```
<path>[::<origin>][{/...|/^}][@[<version-spec>]]
```
where `<version-spec>` is a commitish

**Comment by BenjaminTrapani** [comment]
@jamesgarfield Thank you for pointing that out! I didn't notice that feature. Will use this to test changes in the future. I just built the binary and ran it in gillnet without issue a few minutes ago.

**Comment by jamesgarfield** [comment]
👍

**Comment by BenjaminTrapani** [comment]
@quintenpalmer Here is the ticket: https://clypdinc.atlassian.net/browse/MAIN-19264

**Pull request merged**